src/util/union_find.cpp

```diff
@@ -80,13 +80,13 @@ void unsigned_union_find::re_root(size_type old_root, size_type new_root)
   old_root=find(old_root);
 
   // same set?
-  // assert(find(new_root)==old_root);
+  // assert(find(new_root)==old_root); // Commented out dead code? To review
   if(find(new_root)!=old_root)
```

Comment:
This is basically saying "if the new root is a child of the old root do nothing" which is a bit sketchy